### PR TITLE
[HU#1382][ADD] l10n_pe_base: include SUNAT catalogs

### PR DESCRIPTION
**Pull request by osvalr**
This pull intents to include SUNAT's catalogs to comply with requirements for electronic invoicing
 
- [X] [HU#1382 CA#1](https://www.vauxoo.com/web#id=1382&view_type=form&model=user.story&menu_id=557&action=726) Documents types codes (SUNAT's Catalog no. 1)
> image

- [X] [HU#1382 CA#8](https://www.vauxoo.com/web#id=1382&view_type=form&model=user.story&menu_id=557&action=726) Include code for daily summary for taxable amounts type (SUNAT's Catalog no. 11)
> image

- [X] [HU#1382 CA#13](https://www.vauxoo.com/web#id=1382&view_type=form&model=user.story&menu_id=557&action=726) Tax code types (SUNAT's Catalog no. 5)
> image

- [X] [HU#1382 CA#14](https://www.vauxoo.com/web#id=1382&view_type=form&model=user.story&menu_id=557&action=726) Include credit note reasons (SUNAT's Catalog no. 9)
> image
> image

- [X] [HU#1382 CA#10](https://www.vauxoo.com/web#id=1382&view_type=form&model=user.story&menu_id=557&action=726) Other tributaries concepts codes (SUNAT's Catalog no. 14)
> image

- [X] [HU#1382 CA#4](https://www.vauxoo.com/web#id=1382&view_type=form&model=user.story&menu_id=557&action=726) Tax exemptions reason codes (SUNAT's Catalog no. 7). 
> image
> NOTE: Martín suggested to use only the code `10  Gravada- Operación Onerosa`

- [X] [HU#1382 CA#11](https://www.vauxoo.com/web#id=1382&view_type=form&model=user.story&menu_id=557&action=726) Legends on invoices and other documents (SUNAT's Catalog no. 15). 
> image

- [X] [HU#1382 CA#12](https://www.vauxoo.com/web#id=1382&view_type=form&model=user.story&menu_id=557&action=726) Legends on invoices and other documents (SUNAT's Catalog no. 16). 
> image
> NOTE: Martín suggested that is going to be use only the `01: Precio Unitario (incluye el IGV)`

### CODE REVIEW

File: l10n_pe_base/models/l10n_pe_mixin.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    @api.model
+    def get_document_type(self):
+        """This method is intended to return document type codes based
+        on document object and values on its records.
+        """
+        if self._name == "account.invoice":
+            doctype = (False, False)
+            if (self.type == 'out_invoice' and
+                    self.journal_id.short_code == "FAC"):
```
Comment (osvalr) [inline l10n_pe_base/models/l10n_pe_mixin.py]: I've created one document Factura, another one Boleta de Ventas and the last one Guia de Remisión from   and I see that all of them are pretty the same except for the journal they're using. 
The following is the way I check on them:
```bash
 id | comment |  date_due  |        create_date         | partner_bank_id |    number     | journal_id | amount_total_company_signed | residual | partner_id | create_uid | amount_untaxed | reference | residual_company_signed | amount_total_signed | message_last_post | company_id | amount_tax | state | move_id |    type     | sent | account_id | reconciled | origin |   move_name   | reference_type | date_invoice | payment_term_id |         write_date         | residual_signed |    date    | user_id | write_uid | fiscal_position_id | amount_total | amount_untaxed_signed | currency_id | refund_invoice_id | name | commercial_partner_id | partner_shipping_id | team_id | campaign_id | medium_id | source_id | purchase_id | incoterms_id |      invoice_datetime      |   date_invoice_tz   
----+---------+------------+----------------------------+-----------------+---------------+------------+-----------------------------+----------+------------+------------+----------------+-----------+-------------------------+---------------------+-------------------+------------+------------+-------+---------+-------------+------+------------+------------+--------+---------------+----------------+--------------+-----------------+----------------------------+-----------------+------------+---------+-----------+--------------------+--------------+-----------------------+-------------+-------------------+------+-----------------------+---------------------+---------+-------------+-----------+-----------+-------------+--------------+----------------------------+---------------------
  8 |         | 2017-02-03 | 2017-01-03 23:54:59.106271 |                 | BOL/2017/0001 |         10 |                     40000.0 |  40000.0 |          8 |          1 |        40000.0 |           |                 40000.0 |             40000.0 |                   |          1 |        0.0 | open  |       9 | out_invoice | f    |         30 | f          |        | BOL/2017/0001 | none           | 2017-01-04   |               3 | 2017-01-03 23:55:53.80297  |         40000.0 | 2017-01-04 |       1 |         1 |                    |      40000.0 |               40000.0 |         162 |                   |      |                     8 |                   8 |       1 |             |           |           |             |              | 2017-01-03 23:55:53.818775 | 2017-01-04 00:55:53
 10 |         | 2017-02-03 | 2017-01-03 23:55:24.22419  |                 | BOL/2017/0001 |         12 |                     40000.0 |  40000.0 |          8 |          1 |        40000.0 |           |                 40000.0 |             40000.0 |                   |          1 |        0.0 | open  |       7 | out_invoice | f    |         30 | f          |        | BOL/2017/0001 | none           | 2017-01-04   |               3 | 2017-01-03 23:55:48.374234 |         40000.0 | 2017-01-04 |       1 |         1 |                    |      40000.0 |               40000.0 |         162 |                   |      |                     8 |                   8 |       1 |             |           |           |             |              | 2017-01-03 23:55:48.399482 | 2017-01-04 00:55:48
  9 |         | 2017-02-03 | 2017-01-03 23:55:01.975429 |                 | FAC/2017/0001 |         11 |                     40000.0 |  40000.0 |          8 |          1 |        40000.0 |           |                 40000.0 |             40000.0 |                   |          1 |        0.0 | open  |       8 | out_invoice | f    |         30 | f          |        | FAC/2017/0001 | none           | 2017-01-04   |               3 | 2017-01-03 23:55:51.023661 |         40000.0 | 2017-01-04 |       1 |         1 |                    |      40000.0 |               40000.0 |         162 |                   |      |                     8 |                   8 |       1 |             |           |           |             |              | 2017-01-03 23:55:51.043098 | 2017-01-04 00:55:51
(3 rows)
```

Comment (nhomar) [inline l10n_pe_base/models/l10n_pe_mixin.py]: can you have 2 journals with the same short code?

Comment (osvalr) [inline l10n_pe_base/models/l10n_pe_mixin.py]: @nhomar It seems it's valid to have duplicated short codes. Please take a look at this following issue https://github.com/Vauxoo/superpet/issues/23

Comment (mmendozadelsolar) [inline l10n_pe_base/models/l10n_pe_mixin.py]: @nhomar yes, for example, 2 journals of factura(Store 1 and Store 2) with different sequence. Both are "FAC"

**Comment by osvalr** [comment]
@nhomar could you please review this pull and give me your comments about it.

**Pull request merged**